flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/operations/ddl/AlterTableCompactOperation.java

```diff
@@ -18,41 +18,32 @@
 
 package org.apache.flink.table.operations.ddl;
 
-import org.apache.flink.table.catalog.CatalogPartitionSpec;
 import org.apache.flink.table.catalog.ObjectIdentifier;
-import org.apache.flink.table.operations.OperationUtils;
+import org.apache.flink.table.catalog.ResolvedCatalogTable;
+import org.apache.flink.table.operations.CatalogQueryOperation;
 
-import javax.annotation.Nullable;
-
-import java.util.Collections;
-import java.util.LinkedHashMap;
 import java.util.Map;
 
 /** Operation to describe "ALTER TABLE [PARTITION partition_spec] COMPACT" statement. */
-public class AlterTableCompactOperation extends AlterTableOperation {
+public class AlterTableCompactOperation extends CatalogQueryOperation {
 
-    private final CatalogPartitionSpec partitionSpec;
+    private final ResolvedCatalogTable resolvedManagedTable;
+    private final Map<String, String> compactOptions;
 
     public AlterTableCompactOperation(
```

Comment:
Maybe we can drop this class, just add `dynamicOptions` to `CatalogQueryOperation`.

Reply:
If so, how can we tell the normal `CatalogQueryOperation` from `ALTER TABLE COMPACT` during the rel conversion in`QueryOperationConverter` (e.g. check `isBatch` condition and inject dynamic options)?